docs/index.asciidoc

```diff
@@ -63,8 +65,20 @@ output plugins.
 Can Retry.
 Setting this value true helps user to send multiple retry attempts if the first request fails
 
+[id="plugins-{type}s-{plugin}-flush_size"]
+===== `flush_size`
+
+  * Value type is <<number,number>>
+  * Default value is 100
+
+This plugin uses the bulk index api for improved indexing performance.
+To make efficient bulk api calls, we will buffer a certain number of
```

Comment:
My concern with buffering data is that we lose control of acknowledgement for these events. All plugins tend to process data in a synchronous fashion as this is the only want to acknowledge that the events were processed correctly (or not).

If, for example, the persistent queue is enabled and this plugin accepts a few events, buffers them, then at some point tries to write to loggly but logstash is terminated for some reason, all buffered events are lost since, from the PQ's perspective, they were processed.

As an alternative I suggest moving to the same strategy as the elasticsearch output, that uses a multi event api `multi_receive(events)`: https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/71b2e530873728e60355e02adb3a9e7fa8c660e1/lib/logstash/outputs/elasticsearch/common.rb#L34-L37